spring-integration-jms/src/main/java/org/springframework/integration/jms/dsl/JmsPollableMessageChannelSpec.java

```diff
@@ -35,11 +35,12 @@
  *
  * @author Artem Bilan
  * @author Gary Russell
+ * @author Artem Vozhdayenko
  *
  * @since 5.0
  */
-public class JmsPollableMessageChannelSpec<S extends JmsPollableMessageChannelSpec<S>>
-		extends MessageChannelSpec<S, AbstractJmsChannel> {
+public class JmsPollableMessageChannelSpec<S extends JmsPollableMessageChannelSpec<S, T>, T extends AbstractJmsChannel>
```

Comment:
Yeah... I see now your pain: the `SubscribableJmsChannel` is not a `PollableJmsChannel`, so their specs really must not extend each other.

Let's really revise this flaw in the next `6.0`!

Merging your fix so far...